MaxText/inference_microbenchmark.py

```diff
@@ -277,11 +287,17 @@ def main(config):
       config, engine, params, decode_state, engine.max_concurrent_decodes, cache_size, model_size, benchmark_loop_iters)
 
   results = collate_results(config, benchmark_results, model_size, cache_size, num_model_params)
-  write_results(results, filename=config.inference_microbenchmark_log_file_path)
   print_results_for_analyze(results)
+  flatten_microbenchmark_results = pyconfig.string_to_bool(inference_metadata.get('flatten_microbenchmark_results', 'false'))
+  results = write_results(
+    results,
+    filename=config.inference_microbenchmark_log_file_path,
+    flatten_microbenchmark_results=flatten_microbenchmark_results
+  )
+  return results
 
 
 if __name__ == "__main__":
   jax.config.update("jax_default_prng_impl", "unsafe_rbg")
   pyconfig.initialize(sys.argv)
-  main(pyconfig.config)
+  main(pyconfig.config, {})
```

Comment:
Do we pass in inference_metadata or just a {}?

Reply:
I see, we probably can do
```def main(config, inference_metadata):```
and keep the ```main(pyconfig.config)```

Reply:
Done.